src/lib/common/include/sol-certificate.h

```diff
@@ -84,6 +88,20 @@ const char *sol_cert_get_filename(const struct sol_cert *cert);
 struct sol_blob *sol_cert_get_contents(const struct sol_cert *cert);
 
 /**
+ * @brief Write @a contents to @a cert.
+ *
+ * Certificates are always saved in user context.
+ *
+ * @param file_name The name of the certificate file. The certificate file name
+ *        must be relative. File name with full path is not supported.
+ * @param contents A blob containing the contents to be written to the
+ *        certificate.
+ *
+ * @return 0 on success or a negative error number on errors.
+ */
+int sol_cert_write_contents(const char *file_name, const struct sol_blob *contents);
```

Comment:
remove the `const` from blob, otherwise you can't take a reference to it. If there is no need to keep a reference in any of the API, then using a `struct sol_str_slice contents` is enough.